src/Lib/Effects/Measure.hs

```diff
@@ -36,10 +36,10 @@ timedActionImpl
        )
     => m a -> m a
 timedActionImpl action = do
-    start <- liftIO getCPUTime
+    start <- liftIO (getTime Monotonic)
     !result <- action
-    end <- liftIO getCPUTime
-    let !timeTaken = fromIntegral (end - start) * 1e-12
+    end <- liftIO (getTime Monotonic)
+    let !timeTaken = fromIntegral (sec end - sec start) * 1e-12
```

Comment:
Previous `getCPUTime` returned time in picoseconds. That's why it was multiplied by `1e-12`. Now we don't need this multiplication. Instead it's better to conver time to nanoseconds using `toNanoSecs` function and multiply by `1e-9`